### PR TITLE
fix `Page throws an error when handler passed to the addEventListener function equals undefined` (close #1251)

### LINKED ISSUE #1251: Page throws an error when handler passed to the addEventListener function equals undefined
The error does not occurs on page without `hammerhead`.
```javascript
window.addEventListener("message", void 0);
```
Page: http://zaycev.net/

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev  @churkin

### CODE REVIEW

File: src/client/sandbox/event/listeners.js [hunk 1]
```diff
@@ -87,6 +87,11 @@ export default class Listeners extends EventEmitter {
         return true;
     }
 
+    static _isCorrectListener (listener) {
```
Comment (miherlosev) [inline src/client/sandbox/event/listeners.js]: Rename to the `_isValidEventListener (listener)`

File: src/client/sandbox/event/listeners.js [hunk 2]
```diff
@@ -87,6 +87,11 @@ export default class Listeners extends EventEmitter {
...
+        return listener && (typeof listener === 'function' ||
```
Comment (miherlosev) [inline src/client/sandbox/event/listeners.js]: `return typeof listener === 'function' || eventUtils.isObjectEventListener(listener)`;

File: test/client/fixtures/sandbox/event/event-test.js [hunk 3]
```diff
@@ -301,3 +302,21 @@ test('SVGElement.dispatchEvent should be overriden (GH-614)', function () {
 
     svg.parentNode.removeChild(svg);
 });
+
+test('should not wrapped invalid handlers (GH-1251)', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/event/event-test.js]: test('should not wrap invalid event handlers (GH-1251)`

File: test/client/fixtures/sandbox/event/event-test.js [hunk 4]
```diff
@@ -301,3 +302,21 @@ test('SVGElement.dispatchEvent should be overriden (GH-614)', function () {
...
+    var handlers = [void 0, 1, null, 'str', {}];
+
+    var testHandlers = function (obj) {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/event/event-test.js]: `var testHandlers = function (target) {`

File: src/client/utils/event.js [hunk 5]
```diff
@@ -53,7 +53,7 @@ export function stopPropagation (ev) {
 }
 
 export function isObjectEventListener (listener) {
-    return typeof listener === 'object' && typeof listener.handleEvent === 'function';
+    return listener && typeof listener === 'object' && typeof listener.handleEvent === 'function';
```
Comment (miherlosev) [inline src/client/utils/event.js]: First check needs only for preventing error on access to the `listener.handleEvent` property then `listener` is null.
Let's rephrase as 
`return typeof listener === 'object' && typeof (listener && listener.handleEvent) === 'function'`;

File: test/client/fixtures/sandbox/event/event-test.js [hunk 6]
```diff
@@ -301,3 +302,31 @@ test('SVGElement.dispatchEvent should be overriden (GH-614)', function () {
 
     svg.parentNode.removeChild(svg);
 });
+
+test('should not wrap invalid event handlers (GH-1251)', function () {
+    var handlers = [void 0, 1, null, 'str', {}];
+
+    // NOTE: some handlers in some browsers throw the "Invalid argument" error
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/event/event-test.js]: Don't exclude handlers that raise exception from testing.
Let's change test scenario to:
- memory handlers that raises an error (nativeAddEventListener)
- add handler to overriden addEventListener. If it raises an error then try to find approprite error in stored errors. If it called without error then check the `listeningCtx`.

Also add `Symbol` type checking (if browser supports this type)

File: src/client/utils/event.js (hunk 5 above)
Comment (LavrovArtem) [reply src/client/utils/event.js]: Then `return typeof listener === 'object' && listener && listener.handleEvent === 'function';`

Comment (miherlosev) [reply src/client/utils/event.js]: I don't understand what do you mean

Comment (LavrovArtem) [reply src/client/utils/event.js]: Sorry I am confused, that's right variant:
`return typeof listener === 'object' && listener && typeof listener.handleEvent === 'function';`

Comment (miherlosev) [reply src/client/utils/event.js]: Ok

**Comment by LavrovArtem** [comment]
FPR

File: test/client/fixtures/sandbox/event/event-test.js [hunk 7]
```diff
@@ -301,3 +302,44 @@ test('SVGElement.dispatchEvent should be overriden (GH-614)', function () {
...
+        catch (e) {
+            if (!errorText)
+                errorText = e.toString();
+
+            invalidHandlers.push(handlers[i]);
+        }
+    }
+
+    var testHandlers = function (target) {
+        var storedHandlersCount = listeningCtx.getEventCtx(target, 'click').wrappers.length;
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/event/event-test.js]: `storedHandlerWrappersCount`

File: test/client/fixtures/sandbox/event/event-test.js [hunk 8]
```diff
@@ -301,3 +302,44 @@ test('SVGElement.dispatchEvent should be overriden (GH-614)', function () {
...
+
+test('should not wrap invalid event handlers (GH-1251)', function () {
+    var handlers        = [void 0, 1, null, 'str', {}];
+    var invalidHandlers = [];
+    var errorText       = null;
+
+    if ('Symbol' in window)
+        handlers.push(Symbol('foo'));
+
+    // NOTE: some handlers in some browsers throw the "Invalid argument" error
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/event/event-test.js]: // NOTE:  on adding some type of handlers an "Invalid argument" error can be raised

**Pull request merged**